### PR TITLE
Priya

### PR DESCRIPTION
**Pull request by slpriya**
Homepage route added.
Deleted Is public user logic in likes and unlikes

### CODE REVIEW

**Review by Emily484 — APPROVED** [review]
Looks good to merge

File: routes/api/homepage.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const User = require('../../models/User');
+
+router.get('/home', passport.authenticate('jwt', { session: false }), (req, res) => {
+  var userFollowing = [];
+  var myPost = mongoose.Types.ObjectId(req.user.id);//get logged in user id to get post
+  
+  User.findById(req.user.id, ('following'))
+  .populate('following.user').lean()
+  .then(following => {
+    if (following) {
```
Comment (solongoch) [inline routes/api/homepage.js]: no need of if condition, because "following" would always be truthy(req.user.id)

File: routes/api/homepage.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const passport = require('passport');
+const Post = require('../../models/Posts');
+const User = require('../../models/User');
+
+router.get('/home', passport.authenticate('jwt', { session: false }), (req, res) => {
+  var userFollowing = [];
+  var myPost = mongoose.Types.ObjectId(req.user.id);//get logged in user id to get post
```
Comment (solongoch) [inline routes/api/homepage.js]: What is the difference between myPost and req.user.id? They both hold the same value.

Also, 'myPost' variable name is confusing, myId or myUserId might be good.

File: routes/api/homepage.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  
+  User.findById(req.user.id, ('following'))
+  .populate('following.user').lean()
+  .then(following => {
+    if (following) {
+      userFollowing = following.following.map(follow => follow.user._id);
+    } 
+    //Pushing userid of logged in user into userFollowing[] to fetch posts
+        userFollowing.push(myPost);
+        console.log(userFollowing);
```
Comment (solongoch) [inline routes/api/homepage.js]: we don't need console.log here

File: routes/api/homepage.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    //Pushing userid of logged in user into userFollowing[] to fetch posts
+        userFollowing.push(myPost);
+        console.log(userFollowing); 
+        //
+        Post.find({ "postedBy": { "$in": userFollowing } } ).lean()
+            .populate('postedBy likes.likedBy comments.commentedBy ',
+              ['name', 'username', 'avatar'])
+            .sort({ timePosted: -1 })//for latest records            
+            .exec((err, records) => {
+              if(records) {
```
Comment (solongoch) [inline routes/api/homepage.js]: what will happen if there's no record?

**Review by solongoch — CHANGES_REQUESTED** [review]
there are comments in homepage.js

File: routes/api/homepage.js (hunk 2 above)
Comment (slpriya) [reply routes/api/homepage.js]: I removed variable

File: routes/api/homepage.js (hunk 3 above)
Comment (slpriya) [reply routes/api/homepage.js]: done

File: routes/api/homepage.js (hunk 2 above)
Comment (slpriya) [reply routes/api/homepage.js]: removed the req.user.id from the array.

File: routes/api/homepage.js (hunk 4 above)
Comment (slpriya) [reply routes/api/homepage.js]: Let the comments be there in homepage.js. There is another way to do the above logic. so I commented it

Comment (slpriya) [reply routes/api/homepage.js]: removed

**Comment by slpriya** [comment]
Creating a new one

**Pull request closed without merging**